Firestore/Source/Local/FSTLevelDBRemoteDocumentCache.mm

```diff
@@ -84,6 +84,26 @@ - (nullable FSTMaybeDocument *)entryForKey:(const DocumentKey &)documentKey {
   }
 }
 
+- (MaybeDocumentMap)entriesForKeys:(const DocumentKeySet &)keys {
+  MaybeDocumentMap results;
+
+  LevelDbRemoteDocumentKey currentKey;
+  auto it = _db.currentTransaction->NewIterator();
+
+  for (const DocumentKey &key : keys) {
+    if (key > currentKey.document_key()) {
+      it->Seek([self remoteDocumentKey:key]);
+    }
+    if (!it->Valid() || !currentKey.Decode(it->key()) || currentKey.document_key() != key) {
```

Comment:
Our docs claim that if currentKey.Decode() fails, then the instance is left in an undefined state (until the next call to Decode()). So although this probably works, the next call to document_key() on line 94 isn't guaranteed to do what you want.

Reply:
Could you clarify when decode key will fail? If we believe it only fails when something corrupts, I'd prefer let the code abort rather than continue and hide until later.

Reply:
Rich, thanks for spotting it. I removed that line, so this is no longer applicable.

Zhi, my understanding is this: `Seek` will return an iterator that points either at the key that is being sought, or the first key that is larger. However, in leveldb all the keys are stored in a single table, so it's possible that `Seek` will return a key from the next "logical" table. For example, imagine that keys go like this:
```
remote_documents/foo
remote_documents/bar
write_batches/spam
```
and we're seeking for `remote_documents/zzz`. `Seek` will return an iterator to `write_batches/spam`, which is from a different logical table (by logical, I mean that Firestore uses prefixes like "remote_documents" to treat leveldb as if it had several tables). Then `LevelDbRemoteDocumentKey` will fail to `Decode` because it expects table name `remote_documents`, but it's actually `write_batches`.